### ISSUE TITLE
Scroll into view

### ISSUE THREAD

**Issue by pietermartin** [PepsRyuu/virtual-scrolling-tree#4]
Hi,
Some background. I am adding keyboard navigation support for up/down/left/right to the tree.
The basics I have managed to do. However when using the left key on a leaf or collapsed node, I'd like to put the focus on the parent element.
Not sure how to get this right if the parent is not in the view port.

Think we need some mechanism to scroll an item into the view port.

**Comment by PepsRyuu** [comment]
Hi @pietermartin 

It should be possible to add a function to scroll to an item as long as its parent tree has been expanded. Let me see what I can do and I'll get back to you!

**Comment by pietermartin** [comment]
Thanks, perhaps while you at it, a `postItemsRender` hook will also be nice.
My use case is to use it to set the `focus` on a particular item after all the items has rendered.

**Comment by PepsRyuu** [comment]
Will have to look into that another time. Feel free to write a separate issue on it.

In terms of scrolling functionality, I have a rough implementation but there's quite a few bugs. Might take some time to figure out this fully. It's getting somewhere though.

**Comment by PepsRyuu** [comment]
I created a branch with a new ```scrollIntoView``` function. https://github.com/PepsRyuu/virtual-scrolling-tree/tree/ScrollIntoView

Functions exactly the same as ```expand``` and ```collapse```. You need to pass in an object with the item parent, id, offset etc. There's also a second optional parameter for the function where you can define how to align the item when it's scrolled to.

```
tree.scrollIntoView({
    id: 1
    parent: 0,
    offset: 5,
    children: 5
}, {
    align: "start" | "end"
});
```

Let me know if this works for your use case! 😊

**Comment by pietermartin** [comment]
Great tx, I'll try it out some time today.

**Comment by pietermartin** [comment]
Got some time to test it. Working great, thanks.

**Comment by PepsRyuu** [comment]
Great! I'll release it as soon as I can. :)

**Comment by PepsRyuu** [comment]
Released in 0.5.1
